logger.py

```diff
@@ -23,8 +23,7 @@ def error_and_exit(self: 'ExtendedLogger', msg: str, *, exit_code: int = -1) ->
         sys.exit(exit_code)
 
 
-def configure_logger(lvl: Optional[int] = None) -> ExtendedLogger:
-    logger = logging.getLogger("CDA")
+def configure_logger(lvl: Optional[int | str] = None) -> ExtendedLogger:
```

Comment:
After some discussion with Balazs, I think it makes more sense to only set this via the arguments handed to CDA and not via an environment variable.

Setting via an environment variable was added to support setting the logging when running pytests, however this should always be set to DEBUG anyways. This can also be set in one line prior to running ad hoc commands in a python shell.

Let's avoid setting this in multiple places, as it comes at the cost of clarity in the code, seeing as this function is no longer explicit in its use
i.e.

```
- configure_logger(lvl: int)
+ configure_logger(lvl: Optional[int | str] = None)
```

We should revert #152 and instead hardcode the logger level in all pytest files moving forward.